src/formatting.ts

```diff
@@ -45,6 +46,74 @@ function toggleCodeBlock() {
     return editor.insertSnippet(new SnippetString('```$0\n$TM_SELECTED_TEXT\n```'));
 }
 
+enum QuoteState {
+    // State 1: part of lines has been quoted, need to be quoted
+    PARTIALQUOTED,
```

Comment:
I think it is more natural to quote all the lines (not only the unquoted lines) in this case (thinking about doing multi-line comment).

Reply:
In consideration of the selected text not always accurate, especially like i use three finder drag, i may select more lines which have been quoted. In this situation, what i need is just to quote the unquoted lines. There is no need to quote which have been quoted which may cause the text format disturbed. So could you share the example which is preferred to quote all selected lines?

Reply:
> In consideration of the selected text not always accurate, especially like i use three finder drag, i may select more lines which have been quoted. In this situation, what i need is just to quote the unquoted lines.

I agree this is probably a good design (for some people, or maybe the majority). But as a first step, I would like to make it consistent with other similar commands, e.g.

![multi-line](https://user-images.githubusercontent.com/7588612/93001524-2c79c780-f562-11ea-99b8-a88f88e12be2.gif)

Reply:
Maybe add a switch to decide wthether need to only quote unquoted lines so that default quote behavior is the same as others?

Reply:
@Sean10 can't the user just solve your case by hitting the unquoting command then hitting the quoting command? Same result.

I agree with @yzhang-gh there are many more instances when it's preferable to nest blockquotes.